tasks/connectors/netsparker/lib/netsparker_client.rb

```diff
@@ -23,6 +23,23 @@ def get_last_scan_vulnerabilities(schedule_id)
           JSON.parse(response)
         end
 
+        def receive_all_schedule_ids
+          page = 1
+          schedule_ids = []
+          loop do
+            response = http_get(list_scheduled_url(page), @headers)
+            scheduled_scan_result = JSON.parse(response)
+            schedule_ids.push(*scheduled_scan_result.fetch("List").map { |scan| scan.fetch("Id") }.uniq)
+            break if scheduled_scan_result["IsLastPage"]
+
+            page += 1
+          end
+          schedule_ids.uniq
```

Comment:
Why is `uniq` called again? Is called in the loop also.

Reply:
My bad. It should just be called once outside of the loop